src/03-deals/quotations.apib

```diff
@@ -156,4 +156,41 @@ Download a quotation in a specific format.
     + Attributes (object)
         + data (object)
             + type: `quotation` (string)
-            + id: `e4bf74ca-e900-471f-84b1-276e5d3afae4` (string)
\ No newline at end of file
+            + id: `e4bf74ca-e900-471f-84b1-276e5d3afae4` (string)
+
+### quotations.update [POST /quotations.update]
+
++ Request (application/json)
+
+    + Attributes (object)
+        + quotation_id: `5b16f6ee-e302-0079-901b-50c26c4a55b1` (string, required)
+        + `grouped_lines` (array, optional)
+            + (object)
+                + section (object, optional)
+                    + title (string, required)
+                + line_items (array, required)
+                    + (object)
+                        + quantity: `3` (number, required)
+                        + description: `An awesome product` (string, required)
+                        + extended_description: `Some more information about this awesome product` (string, optional) - Uses Markdown formatting
+                        + unit_price (object, required)
+                            + Include Money
+                            + tax: `excluding` (enum, required)
+                                + Members
+                                    + excluding
+                        + tax_rate_id: `c0c03f1e-77e3-402c-a713-30ea1c585823` (string, required)
+                        + discount (object, optional)
+                            + value: 10 (number, required)
+                            + type (enum, required)
+                                + Members
+                                    + percentage - Values between 0 and 100
+        + discounts (array, optional)
+            + (object)
+                + value: 10 (number, required)
+                + type (enum, required)
+                    + Members
+                        + percentage - Values between 0 and 100
+                + description: `winter promotion` (string, required)
+        + text: `Quotation text` (string, optional) - Uses Markdown formatting. A quotation needs `grouped_lines` and/or `text` to be valid
```

Comment:
The text property should take `null` to allow it to be unset. (Not providing it will not update the property.)
```suggestion
        + text: `Quotation text` (string, optional, nullable) - Uses Markdown formatting. A quotation needs `grouped_lines` and/or `text` to be valid
```

Reply:
Allowing this will require extra validation so you don't end up without product lines and no text, should we do that for the first version?

Reply:
The business rule of having either grouped lines or text or both indeed still needs to stand, and we should also not disallow to remove a text from a quotation again through the update endpoint. The latter would be pretty weird for an integrator.

So I think this is some essential behaviour to publicly release the endpoint.

(Basically we already need similar validation rules as passing an empty string for a text would create the same behaviour as passing `null`.)